### ISSUE TITLE
[Bug] iOS broker redirect uri validation

### ISSUE THREAD

**Issue by last1here** [AzureAD/microsoft-authentication-library-for-dotnet#1995]
**Which Version of MSAL are you using ?**
MSAL 4.17.1

**Platform**
Xamarin Forms (iOS)

**What authentication flow has the issue?**
* Desktop / Mobile
    * [x] Interactive
    * [ ] Integrated Windows Auth
    * [ ] Username Password
    * [ ] Device code flow (browserless)
* Web App
    * [ ] Authorization code
    * [ ] OBO
* Web API
    * [ ] OBO

**Is this a new or existing app?**
Existing app upgrading MSAL from 4.16.1 to 4.17.1 and adding broker support.

**Repro**

```csharp
PCA = PublicClientApplicationBuilder.Create(authClientId)
                .WithTenantId(authTenantId)
                .WithIosKeychainSecurityGroup(authKeychain)
                .WithBroker()
                .WithRedirectUri($"msauth.{bundleId}://auth")
                .Build();
```

**Expected behavior**
The brokered flow completes correctly.

**Actual behavior**
The `MsalError.CannotInvokeBroker` exception thows. Works as expected in 4.16.1.

**Possible Solution**
From a quick check I think that the validation in the iOSBroker class is failing, specifically [this line](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/master/src/client/Microsoft.Identity.Client/Platforms/iOS/Broker/iOSBroker.cs#L115) added as part of #1929 between 4.16.1 and 4.17.0. The `redirectUri.AbsoluteUri` returns the Uri with a trailing `/` which is then compared to expectedRedirectUri which is missing the trailing `/`. Adding a trailing `/` to expectedRedirectUri should resolve the issue.

**Comment by bgavrilMS** [comment]
Thank you for reporting and investigating @last1here , I'll get this fixed ASAP.

**Comment by markbeij** [comment]
We have this issue as well. This should be patched to 4.17.2 as this makes MSAL for iOS quite useless

**Comment by trwalke** [comment]
fixed in [Release 4.18.0](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/releases/tag/4.18.0)
